src/iterator.cr

```diff
@@ -1093,13 +1093,7 @@ module Iterator(T)
       if @tail
         tail_key, tail_val = @tail_key, @tail_val
         @tail = false
-        case tail_key
-        when nil, :_separator
-        when :_alone
-          return {tail_key.not_nil!, [tail_val]}
-        else
-          raise ArgumentError.new("symbols beginning with an underscore are reserved")
-        end
+        return {tail_key.not_nil!, [tail_val]} if tail_key == Enumerable::Chunk::Alone
```

Comment:
Is this `not_nil!` needed now? Could you maybe add a spec with `nil` values? :-)